tempy/tempy.py

```diff
@@ -402,6 +406,51 @@ def wrap(self, other):
         other.append(self)
         return self
 
+    def wrap_many(self, *args, strict=False):
+        """Wraps different copies of this element inside all empty tags
+        listed in params or param's (non-empty) iterators.
+
+        Returns list of copies of this element wrapped inside args
+        or None if not succeeded, in the same order and same structure,
+        i.e. args = (Div(), (Div())) -> value = (A(...), (A(...)))
+
+        If on some args it must raise TagError, it will only if strict is True,
+        otherwise it will do nothing with them and return Nones on their positions"""
+        wcopies = []
+        failure = []
+
+        def wrap_next(tag, idx):
+            nonlocal wcopies, failure
+            next_copy = self.__copy__()
+            try:
+                return next_copy.wrap(tag)
+            except TagError:
+                failure.append(idx)
+                return None
+
+        arg_idx = -1
+        for arg in args:
+            arg_idx += 1
+            if isinstance(arg, DOMElement):
+                wcopies.append(wrap_next(arg, arg_idx))
+            else:
+                if not isinstance(arg, Iterable) or not arg or not isinstance(iter(arg).__next__(), DOMElement):
+                    raise WrongArgsError(self,
+                                         'Argument no {} is not DOMElement nor iterable of DOMElements'.format(arg_idx))
+                else:
+                    iter_idx = -1
+                    iter_wcopies = []
+                    for t in iter(arg):
+                        iter_idx += 1
+                        iter_wcopies.append(wrap_next(t, (arg_idx, iter_idx)))
+                    wcopies.append(type(arg)(iter_wcopies))
+
+        if failure and strict:
+            raise TagError(self, 'Wrapping in a non empty Tag is forbidden, failed on arguments ' +
+                           ', '.join(list(map(lambda idx: str(idx) if type(idx) is int else '[{1}] of {0}'.format(*idx),
```

Comment:
`type(idx) is int`is deprecated, plus this can be simpler if `failures` always contains tuples (using `(arg_idx, 0)` for the first `wrap_next` call in the method) .

Reply:
Thank you for advices! I'l try to fix everything soon :)